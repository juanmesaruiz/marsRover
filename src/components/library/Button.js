import React from 'react';
import styled from 'styled-components';

const Button = ({ children, ...otherProps }) => (
  <button {...otherProps}>{children}</button>
);

const StyledButton = styled(Button)`
  cursor: pointer;
`;

export default StyledButton;
