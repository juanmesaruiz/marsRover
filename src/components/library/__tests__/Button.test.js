import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import Button from '../Button';

describe('Button test suite', () => {
  const component = <Button />;

  const setup = () => render(component);

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Should match snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
