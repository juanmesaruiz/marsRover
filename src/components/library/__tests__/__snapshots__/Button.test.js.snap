// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test suite Should match snapshot 1`] = `
.c0 {
  cursor: pointer;
}

<div>
  <button
    class="c0"
  />
</div>
`;
